### PR TITLE
Condor cli

### PR DESCRIPTION
**Pull request by amikaili**
Enable Starting End to end testing of the new RM

### CODE REVIEW

File: deploy.cfg [hunk 1]
```diff
@@ -21,7 +21,7 @@ ref.data.base=/kb/data
 
 
 condor.mode=1
-condor.submit.desc.file.path=../scripts/job_desc.sub
+condor.submit.desc.file.path=/Users/amikaili/myKbaseCode/njs_wrapper/scripts/
```
[the remark this answers is not in the document]
Comment (amikaili) [reply deploy.cfg]: This is adjusted at deploy time (or in our pending CI testing)

File: src/us/kbase/common/utils/CondorUtils.java [hunk 2]
```diff
@@ -186,14 +193,79 @@ public static float submitToCondorCLI ( String ujsJobId, String selfExternalUrl,
...
+		// Append executable command and query command
+        template += "\n executable = " + submitFilePath + "run_async_srv_method.sh ";
+        // template += "\n executable = " + "/Users/amikaili/myKbaseCode/njs_wrapper/scripts/run_async_srv_method.sh ";
+        template += ujsJobId + " " + selfExternalUrl + " \n";
+        template += "\n query \n";
+        
+        // Write submit_async.sub containing String type::: template, to the scripts dir
+        try {
+            byte bWrite [] = template.trim().getBytes();
+            OutputStream os = new FileOutputStream( submitFilePath + "submit_async.sub");
```
[the remark this answers is not in the document]
Comment (amikaili) [reply src/us/kbase/common/utils/CondorUtils.java]: 'submit_async.sub' file is the condor submit file (is passed as the parm to condor_submit CLI system call) please see condor_submit.sh Also: it's intended to be ephemeral, that is it gets re-written for every job, and no, it shouldn't grow (it's not even saved anywhere)!

File: src/us/kbase/common/utils/CondorUtils.java [hunk 3]
```diff
@@ -186,14 +193,79 @@ public static float submitToCondorCLI ( String ujsJobId, String selfExternalUrl,
...
+            stream.flush();
+
+            os.close();
+        
+            // Echo it back out: check to see what went in the file:
+            InputStream is = new FileInputStream( submitFilePath + "submit_async.sub");            
+            // InputStream is = new FileInputStream("/Users/amikaili/myKbaseCode/njs_wrapper/scripts/submit_async.sub");            
+            int size = is.available();
+            for( int i = 0; i < size; i++ ) {
+               System.out.print( (char) is.read() );
```
[the remark this answers is not in the document]
Comment (amikaili) [reply src/us/kbase/common/utils/CondorUtils.java]: OK I agree with this one!  I will implement this code structure "improvement", if our underlying CLI approach flies?  This needs to happen before we continue to "build" on top of this approach, for LOE sake?  Please see the comments about the latter on the slack channel @tian?

**Comment by amikaili** [comment]
From Slack (and I totally agree):

tian [3:39 PM]
@Arman yea I think if we are still just testing theory. I think that’s fine.

scanon [6:48 PM]
...
I think what Tian raised is trivial to fix.

**Comment by sychan** [comment]
@amikaili If you are asking me to perform the merge, then just ask in clear direct language. Are you under the impression that I am somehow blocking your merge? I *seems* that way, but I cannot be sure, since this is the first vague indication that you are asking that I merge it.

It would help productivity if you paid more attention to being clear and unambiguous in your communications.

**Comment by amikaili** [comment]
Thank you @sychan

**Pull request merged**